### PR TITLE
Tx pool intrinsic gas cost check

### PR DESCRIPTION
**Pull request by Vuksan**
# Description

This PR adds an intrinsic gas cost check for adding a transaction to the transaction pool.
Apart from that it also adds minor changes, which can be seen from the commit messages:

- Logs tx execution error instead of printing it to the output
- Removes unused network field from TxPool struct
- Modifies `addImpl` to accept one transaction only

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have tested this code
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

If you are wondering **why** some changes are made, take a look at commit messages.

### CODE REVIEW

File: helper/tests/testing.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	"testing"
+)
+
+func GenerateKeyAndAddr(t *testing.T) (*ecdsa.PrivateKey, types.Address) {
+	t.Helper()
+	key, err := crypto.GenerateKey()
+	assert.NoError(t, err)
+	addr := crypto.PubKeyToAddress(&key.PublicKey)
+	return key, addr
+}
```
Comment (zivkovicmilos) [inline helper/tests/testing.go:17]: The helper package is so underrated for resolving merge conflicts that arise when you add the most basic shared functionality between 2 modules 😎

**Review by zivkovicmilos — APPROVED** [review]
Great work on the research and in code documentation 👍 
It's much more clearer now what is going now and how our txpool is working.
Additionally, it's good that we have a check now that should've existed long ago, but somehow got omitted while other features were pushed through.

Everything looks good to me 💯

**Review by lazartravica — APPROVED** [review]
Excellent job, and nice commit messages, easy to follow through. 🥇

**Review by Kourin1996 — APPROVED** [review]
LGTM 👏
Great fixes and it'll be more clearly understandable

**Pull request merged**